apps/admin/src/components/users/details/consumption/user-consumption-aggregation-option.tsx

```diff
@@ -1,10 +1,39 @@
 "use client";
 
+import React, { useMemo } from "react";
 import { useUserContext } from "@/hooks/user-hook";
-
 import { AggregationOption } from "@energyleaf/ui/components/utils";
+import { AggregationType } from "@energyleaf/lib";
+import { differenceInCalendarDays, differenceInMonths, differenceInWeeks, differenceInYears } from "date-fns";
 
 export default function UserConsumptionAggregationOption() {
     const context = useUserContext();
-    return <AggregationOption onSelectedChange={context.setAggregationType} selected={context.aggregationType} />;
+
+    const availableOptions = useMemo(() => {
+        const startDate = new Date(context.startDate);
+        const endDate = new Date(context.endDate);
+
+        const dayDiff = differenceInCalendarDays(endDate, startDate);
+        const monthDiff = differenceInMonths(endDate, startDate);
+        const weekDiff = differenceInWeeks(endDate, startDate);
+        const yearDiff = differenceInYears(endDate, startDate);
+
+        const options = [AggregationType.RAW];
+
+        if (dayDiff <= 7) options.push(AggregationType.HOUR);
```

Comment:
Man könnte die Berechnung noch ein wenig effizienter machen. Z. B. kann weekDiff >= 2 && monthDiff <= 3 nicht true sein, wenn dayDiff <= 7 ist. Dadurch kann man sich in manchen Fällen dann auch den Aufruf einiger difference Methode sparen (Z. 16-19). Ist die Frage, ob man dadurch überhaupt viel gewinnt. So wie es jetzt ist, ist es auf jeden Fall die mit Abstand lesbarste Implementierung. Könntest du dir evtl. mal anschauen

Reply:
Hab mir das mal angeguckt aber jetzt so gelassen. Denke so ist es am übersichtlichstem